pricegraph/src/orderbook.rs

```diff
@@ -164,7 +163,11 @@ impl Orderbook {
 
         let (base, quote) = (node_index(market.base), node_index(market.quote));
 
-        while let Err(cycle) = ShortestPathGraph::new(&self.projection, quote) {
+        loop {
+            let cycle = match ShortestPathGraph::new(&self.projection, quote) {
+                Ok(_) => break,
+                Err(cycle) => cycle,
+            };
```

Comment:
I liked the previous version more but no strong opinion.

Reply:
Me too. However, I got another weird compiler error, because a `a temporary with access to the immutable borrow is created` in `ShortestPathGraph::new` and then still alive when I try to `self.fill_path` which also requires an immutable borrow.

Happy for suggestions how to work around that.

Reply:
For me the error is more specific (maybe you can update rustc). It says `the immutable borrow might be used here, when that temporary is dropped and runs the destructor for type `std::result::Result<graph::shortest_paths::ShortestPathGraph ...`.
With that I can see why the compiler doesn't like it and why the match helps. It is something that rustc could improve but I don't think there is a better workaround. Maybe I'll make a short reproduction and create an issue for rustc.